sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -101,16 +101,45 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase {
         "pivotType" -> pivotCol.dataType.catalogString))
   }
 
+  def unpivotRequiresAttributes(given: String,
+                                empty: String,
+                                expressions: Seq[NamedExpression]): Throwable = {
+    val types = expressions.groupBy(_.getClass.getSimpleName)
+      .mapValues(exprs => exprs.map(expr => expr.toString.replaceAll("#\\d+", "")).sorted)
+      .mapValues(exprs => if (exprs.length > 3) exprs.take(3) :+ "..." else exprs)
+      .toList.sortBy(_._1)
+      .map { case (className, exprs) => s"$className (${exprs.mkString(", ")})" }
```

Comment:
We should not expose too many internal information in the user-facing error message. Can we just put `expressions.filterNot(_.isInstanceOf[Attribute]).map(_.sql).mkString(", ")`?